diff --git a/tools/releasetools/ota_from_target_files.py b/tools/releasetools/ota_from_target_files.py
index be0e798685..c9399561a8 100755
--- a/tools/releasetools/ota_from_target_files.py
+++ b/tools/releasetools/ota_from_target_files.py
@@ -860,6 +860,9 @@ else if get_stage("%(bcb_dev)s") == "3/3" then
   boot_img = common.GetBootableImage(
       "boot.img", "boot.img", OPTIONS.input_tmp, "BOOT")
 
+  recovery_img = common.GetBootableImage(
+      "recovery.img", "recovery.img", OPTIONS.input_tmp, "RECOVERY")
+
   if HasVendorPartition(input_zip):
     script.ShowProgress(0.1, 0)
 
@@ -874,6 +877,9 @@ else if get_stage("%(bcb_dev)s") == "3/3" then
   common.CheckSize(boot_img.data, "boot.img", target_info)
   common.ZipWriteStr(output_zip, "boot.img", boot_img.data)
 
+  common.CheckSize(recovery_img.data, "recovery.img", target_info)
+  common.ZipWriteStr(output_zip, "recovery.img", recovery_img.data)
+
   device_specific.FullOTA_PostValidate()
 
   if OPTIONS.backuptool:
@@ -1538,6 +1544,7 @@ else if get_stage("%(bcb_dev)s") != "3/3" then
     if d is None:
       include_full_boot = True
       common.ZipWriteStr(output_zip, "boot.img", target_boot.data)
+      common.ZipWriteStr(output_zip, "recovery.img", target_recovery.data)
     else:
       include_full_boot = False
 
@@ -1588,6 +1595,8 @@ else
   if OPTIONS.two_step:
     common.ZipWriteStr(output_zip, "boot.img", target_boot.data)
     script.WriteRawImage("/boot", "boot.img")
+    common.ZipWriteStr(output_zip, "recovery.img", target_recovery.data)
+    script.WriteRawImage("/recovery", "recovery.img")
     print("writing full boot image (forced by two-step mode)")
 
   if not OPTIONS.two_step:
@@ -1596,6 +1605,8 @@ else
         print("boot image changed; including full.")
         script.Print("Installing boot image...")
         script.WriteRawImage("/boot", "boot.img")
+        script.Print("Installing recovery image...")
+        script.WriteRawImage("/recovery", "recovery.img")
       else:
         # Produce the boot image by applying a patch to the current
         # contents of the boot partition, and write it back to the
