Subject: [PATCH] Ignore Usage Bits Warnings
diff --git a/graphics/mapper/2.0/utils/passthrough/include/mapper-passthrough/2.0/Gralloc0Hal.h b/graphics/mapper/2.0/utils/passthrough/include/mapper-passthrough/2.0/Gralloc0Hal.h
index 7369ac759..51ec64c81 100644
--- a/graphics/mapper/2.0/utils/passthrough/include/mapper-passthrough/2.0/Gralloc0Hal.h
+++ b/graphics/mapper/2.0/utils/passthrough/include/mapper-passthrough/2.0/Gralloc0Hal.h
@@ -64,8 +64,9 @@ class Gralloc0HalImpl : public Hal {
 
         const uint64_t validUsageBits = getValidBufferUsageMask();
         if (descriptorInfo.usage & ~validUsageBits) {
-            ALOGW("buffer descriptor with invalid usage bits 0x%" PRIx64,
-                  descriptorInfo.usage & ~validUsageBits);
+            // We don't need this to constantly spam the logs, so just comment it out
+            //ALOGW("buffer descriptor with invalid usage bits 0x%" PRIx64,
+            //      descriptorInfo.usage & ~validUsageBits);
         }
 
         *outDescriptor = grallocEncodeBufferDescriptor(descriptorInfo);
diff --git a/graphics/mapper/2.0/utils/passthrough/include/mapper-passthrough/2.0/Gralloc1Hal.h b/graphics/mapper/2.0/utils/passthrough/include/mapper-passthrough/2.0/Gralloc1Hal.h
index d9beb4f67..ef3c27769 100644
--- a/graphics/mapper/2.0/utils/passthrough/include/mapper-passthrough/2.0/Gralloc1Hal.h
+++ b/graphics/mapper/2.0/utils/passthrough/include/mapper-passthrough/2.0/Gralloc1Hal.h
@@ -84,8 +84,9 @@ class Gralloc1HalImpl : public Hal {
 
         const uint64_t validUsageBits = getValidBufferUsageMask();
         if (descriptorInfo.usage & ~validUsageBits) {
-            ALOGW("buffer descriptor with invalid usage bits 0x%" PRIx64,
-                  descriptorInfo.usage & ~validUsageBits);
+            // We don't need this to constantly spam the logs, so just comment it out
+            //ALOGW("buffer descriptor with invalid usage bits 0x%" PRIx64,
+            //      descriptorInfo.usage & ~validUsageBits);
         }
 
         *outDescriptor = grallocEncodeBufferDescriptor(descriptorInfo);
-- 
2.25.1

